pakrat/progress.py

```diff
@@ -207,14 +207,15 @@ def callback(self, method, *args):
             try: method(self.repo_id, *args)
             except: pass
 
-    def start(self, _file, url, basename, size, text):
+    def start(self, filename=None, url=None, basename=None, size=None, text=None):
         """ Called by urlgrabber when a file download starts.
 
         All we use this for is storing the name of the file being downloaded so
         we can check that it is an RPM later on.
         """
-        self.package = basename
-        self.callback('download_start', _file, url, basename, size, text)
+        if basename:
+            self.package = basename
+            self.callback('download_start', filename, url, basename, size, text)
```

Comment:
Sorry, this was another fix that would happen after the parameter name fix. By setting the defaults to `None`, it allows both yum and urlgrabber to call this function. We have to set those defaults to `None` because the call from `drpm.py` does not pass basename. If drpm.py calls this and we don't check if basename exists, we would set `self.package` to `None`. It's possible that there would never be an issue, but I added this check just in case so we don't ever step on an existing download. If that happened, when `end` gets called it would not know what package it was referencing.